portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -67,4 +82,38 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
       response.sendRedirect("/index.html");
     }
   }
+
+  /** Returns a URL that points to the uploaded file, or null if the user didn't upload a file. */
+  private String getUploadedFileUrl(HttpServletRequest request, String formInputElementName) {
+    BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+    Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
+    List<BlobKey> blobKeys = blobs.get(formInputElementName);
+
+    // User submitted form without selecting a file, so we can't get a URL. (dev server)
+    if (blobKeys == null || blobKeys.isEmpty()) {
+      return null;
+    }
+
+    // Our form only contains a single file input, so get the first index.
+    BlobKey blobKey = blobKeys.get(0);
```

Comment:
What if blobKey.size() > 1? Though blobKey.size() = 1 is guaranteed in the html file, this class is independent to the html, it's better to regard the html the user of this class, so that you should verify the input here.

In team work, the html and java may be implemented by different people.

Reply:
Thx for your advice! I think I should verify whether blobKeys.size() == 1 at the very beginning.